### ISSUE TITLE
Add support for cc-by-nc imagery

### ISSUE THREAD

**Issue by bgirardot** [hotosm/OpenAerialMap#124]
This is needed to support a lot of imagery that is compatible with OSM.

Specifically DG or Planet, they are both cc-by-nc+osm 

Someone said this was coded but not merged.

**Comment by sfkeller** [comment]
Currently the only licence option to choose from (in the upload dialog) is "CC-BY 4.0". I'd also vote for more licenses to choose from, as Blake mentioned above.

In any case, the license name should also be shown in the metadata section of any image detail page.

**Comment by don-vip** [comment]
Could we please have CC-BY-SA and ODbL as possible licences? My town publishes high-definition raw aircraft imagery every two years and uses these licences. I cannot import it right now.

**Comment by sfkeller** [comment]
Just FYI: The lack of choice of licenses is a show stopper for using OAM to many initiatives I'm aware of.
Technically this is AFAIK no big enhancement (for those who know the code :-) ).

(@bgirardot Do you know who's the maintainer of OAM and if there are resources avail. at HOTOSM?)

**Comment by smit1678** [comment]
To chime in here, apologies for the slowness in replying. We've had adding licenses on the backlog for a while but it's been put on the backburner while several developers contributing are in the middle of a refactor to address the many other usability issues there has been feedback on. Our effort has been mainly focused on the new work and not addressing old bugs or issues at the moment: https://github.com/sharkinsspatial/oam-browser-next. 

Previously there was a hurdle regarding the license and if OAM was only going to have a single license but we've opened that up and now we just need to implement a solution for the upload process. Will see if we can get something up in the near-term to help with allow more than one license.

**Comment by Blake-Loveland** [comment]
This is still an issue that hampers use. In this case, imagery under NextView license needs a home for mapping.

**Comment by sfkeller** [comment]
Yes, can confirm that this is still an issue that hampers use of OAM. 
I'm still unsure who at HOT is responsible to allocate (probably quite available) resources on this issue... 
@bgirardot Do you know who to address?

**Comment by smit1678** [comment]
@sfkeller @bgirardot Agreed, this has been under discussion for a while and well known. Development on the current version is quite cumbersome. There's been a lot of work on the new version and was hoping that it would be available soon as I mentioned last month. It's not going to be ready yet so I'm working on a quick fix right now. OAM new development is mostly a volunteer effort at the moment and HOT is trying to help maintain the running version while volunteers are working on the next version. If others want to jump in with pull requests for fixes or changes like this, we can certainly review and work to get them integrated. Thanks for your patience on this issue, will have an update shortly.

**Comment by sfkeller** [comment]
Oh, thanks @smit1678 for your quick answer and engagement!

**Comment by smit1678** [comment]
To update on progress, this has been tested and now deployed on OAM. 🎉 

<img width="666" alt="openaerialmap browser 2018-12-23 10-40-12" src="https://user-images.githubusercontent.com/796838/50386615-3995b980-06e2-11e9-8476-5c9a7cd8ce03.png">

I'm not going to fully close this ticket yet because this is only an upload form fix. Right now the only way to know the license type for imagery is through the API. We haven't exposed this via the frontend yet so going to leave the ticket open and mark this out as the next action here to close out this feature request.

Here is an example for how to make an API call to see the license information. Take the `uuid` which is the file URI and add that as a parameter to the `/meta` API endpoint. You can find the file URI when you download the file (or right click and copy link location on Firefox).

```
https://api.openaerialmap.org/meta?uuid=https://oin-hotosm.s3.amazonaws.com/5c1e06e139ad6f00053fbc43/0/5c1e06e139ad6f00053fbc44.tif
```

**Comment by smit1678** [comment]
> In this case, imagery under NextView license needs a home for mapping.

@Blake-Loveland To come back to this point, as far as I understand, NextView license is not like any of the creative commons licenses and does not allow the sharing of the raw file publicly. A TMS can be created and shared but the original file cannot (unless it's to specific partners). Right now OAM does not allow restricting downloads. For any NextView imagery, we can help support by creating a TMS that is not exposed via OAM frontend or API, or we can upload to Mapbox through HOT's account and make a specific TMS available for mapping.

**Comment by sfkeller** [comment]
Nice. Looks like a christmas release!

**Comment by don-vip** [comment]
Great! Can we please have ODbL too?

**Comment by andrewharvey** [comment]
I'd like to request adding CC BY-NC-SA, I came across some great aerial imagery under this license + OSM tracing waiver and it would be nice to upload to OAM.
